rust/src/settings.rs

```diff
@@ -0,0 +1,237 @@
+        && s.update < 1.
+        && 0. < s.sum + s.update - s.sum * s.update
+        && s.sum + s.update - s.sum * s.update < 1.
+    {
+        Ok(())
+    } else {
+        Err(ValidationError::new("starvation"))
+    }
+}
+
+#[derive(Debug, Validate, Deserialize, Clone, Copy)]
+/// REST API settings
+pub struct ApiSettings {
+    /// The address to which the REST API should be bound.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [api]
+    /// bind_address = "0.0.0.0:8081"
+    /// # or
+    /// bind_address = "127.0.0.1:8081"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN_API__BIND_ADDRESS=127.0.0.1:8081
+    /// ```
+    pub bind_address: std::net::SocketAddr,
+}
+
+#[derive(Debug, Validate, Deserialize, Clone, Copy)]
+/// Mask settings
+pub struct MaskSettings {
+    /// The order of the finite group.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [mask]
+    /// group_type = "Integer"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN_MASK__GROUND_TYPE=Integer
+    /// ```
+    pub group_type: GroupType,
+
+    /// The data type of the numbers to be masked.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [mask]
+    /// data_type = "F32"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN_MASK__DATA_TYPE=F32
+    /// ```
+    pub data_type: DataType,
+
+    /// The bounds of the numbers to be masked.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [mask]
+    /// bound_type = "B0"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN_MASK__BOUND_TYPE=B0
+    /// ```
+    pub bound_type: BoundType,
+
+    /// The bounds of the numbers to be masked.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [mask]
+    /// model_type = "M3"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN_MASK__MODEL_TYPE=M3
+    /// ```
+    pub model_type: ModelType,
+}
+
+impl Default for MaskSettings {
+    fn default() -> Self {
+        Self {
+            group_type: GroupType::Prime,
+            data_type: DataType::F32,
+            bound_type: BoundType::B0,
+            model_type: ModelType::M3,
+        }
+    }
+}
+
+impl Settings {
+    /// Loads and validates the coordinator settings via a configuration file.
+    /// Fails when the loading of the configuration file or its validation failed.
+    pub fn new(path: PathBuf) -> Result<Self, SettingsError> {
+        let settings: Settings = Self::load(path)?;
+        settings.validate()?;
+        Ok(settings)
+    }
+
+    /// Loads the coordinator settings via a configuration file without validation.
+    /// Fails when the loading of configuration file failed.
+    pub fn new_unchecked(path: PathBuf) -> Result<Self, ConfigError> {
```

Comment:
do we need an unchecked `new`? just wondering

Reply:
Actually that's true. I will remove it